### PR TITLE
Add sub-commands for new configuration management APIs

### PR DESCRIPTION
**Pull request by doanac**
This provides new subcommands to interface with the new OTA config management APIs. The commands introduced are:
~~~
$ fioctl devices config
Device configuration

Usage:
  fioctl devices config [command]

Available Commands:
  delete      Delete file from the current configuration
  log         Changelog of device's configuration
  set         Create a secure configuration for the device
~~~

and
~~~
fioctl config -h
Manage configuration common to all devices in a factory

Usage:
  fioctl config [command]

Available Commands:
  log         Changelog of configuration
  set         Create a new factory-wide configuration
~~~

### CODE REVIEW

**Comment by doanac** [comment]
Hey guys - I'm guessing this will be hard to review. However, it won't be too hard to test and try out. I'm hoping to get some feedback on the usability of it. Maybe there's a better approach? I'm open to changes.

FYI - the server changes are all in place to support this.

**Comment by EmbeddedAndroid** [comment]
I'll test it out in a few, thanks @doanac

**Comment by doanac** [comment]
Hey guys. This stuff actually works now, but ergonmics are getting really weird. I'm going to redo how the "devices" sub command is ot have some new "sub" commands under the sub commands. ie:
~~~
fioctl devices list-config <device>-> fioctl devices config list <device>
~~~

**Comment by doanac** [comment]
@mike-scott @EmbeddedAndroid  - I did some refactoring of fioctl over the weekend in hopes these new commands would make more sense. I think this new approach is better.

Following on this PR will one or two more that will add:
~~~
fioctl devices config updates  <device># Show what docker-apps and tags you are on
fioctl devices config updates <device> [--tag master] [--apps shellhttpd]
~~~

and 
~~~
fioctl devices config wireguard <device> # show if its enabled/disabled
fioctl devices config wireguard <device> <enable|disable>
~~~

File: subcommands/config/set.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	"on-changed": ["/usr/bin/touch", "/tmp/npmtok-changed"]
+      },
+      {
+        "name": "A-Readable-Value",
+	"value": "This won't be encrypted and will be visible from the API",
+	"unencrypted": true
+      },
+      {
+        "name": "githubok",
+	"value": "1234"
```
Comment (ricardosalveti) [inline subcommands/config/set.go:55]: There is a mix of tabs and spaces here.

File: subcommands/devices/config_set.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+	"on-changed": ["/usr/bin/touch", "/tmp/npmtok-changed"]
+      },
+      {
+        "name": "A-Readable-Value",
+	"value": "This won't be encrypted and will be visible from the API",
+	"unencrypted": true
+      },
+      {
+        "name": "githubok",
+	"value": "1234"
```
Comment (ricardosalveti) [inline subcommands/devices/config_set.go:62]: Mix of tabs and spaces here as well.

**Pull request merged**